swarm_nl/src/core/prelude.rs

```diff
@@ -247,214 +256,136 @@ impl StreamResponseBuffer {
 #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub(super) enum Rpc {
 	/// Using request-response.
-	ReqResponse { data: Vec<Vec<u8>> },
+	ReqResponse { data: RpcData },
 }
 
 /// The configuration for the RPC protocol.
-pub struct RpcConfig {
-	/// Timeout for inbound and outbound requests.
-	pub timeout: Duration,
-	/// Maximum number of concurrent inbound + outbound streams.
-	pub max_concurrent_streams: usize,
+pub enum RpcConfig {
+	Default,
+	Custom {
+		/// Timeout for inbound and outbound requests.
+		timeout: Duration,
+		/// Maximum number of concurrent inbound + outbound streams.
+		max_concurrent_streams: usize,
+	},
 }
 
-/// The high level trait that provides an interface for the application layer to respond to network
-/// events.
-pub trait EventHandler {
+/// Enum that represents the events generated in the network layer
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub enum NetworkEvent {
```

Comment:
Note: I'll add a commit to this PR to make the docs updates so we also document the variant fields for completeness.